consensus/tests/integration_test.rs

```diff
@@ -1,442 +1,139 @@
-    // Expected: No operation.
-    let serve_task = tokio::spawn(async { server_node.serve(3_000).await });
-    for other_node in other_nodes.iter_mut() {
-        other_node.fetch().await.unwrap();
-        let timestamp = get_timestamp();
-        let result = other_node.progress(timestamp).await.unwrap_err();
+    let serve_task = tokio::spawn(async move {
+        let task = tokio::spawn(Dms::serve(server_node.get_dms(), server_network_config));
+        sleep_ms(5000).await;
+        task.abort();
+        let _ = task.await;
+        server_node.update().await.unwrap();
+        server_node.progress(0).await.unwrap();
         assert_eq!(
-            result.to_string(),
-            "operation on finalized state".to_string()
+            server_node
+                .check_finalized()
+                .await
+                .unwrap()
+                .unwrap()
+                .block_hash,
+            block_hash
         );
+    });
+
+    async fn sync(client_nodes: &mut [(Consensus, ClientNetworkConfig)]) {
+        for (node, network_config) in client_nodes.iter_mut() {
+            node.flush().await.unwrap();
+            dms::DistributedMessageSet::broadcast(node.get_dms(), network_config)
+                .await
+                .unwrap();
+        }
+        for (node, network_config) in client_nodes.iter_mut() {
+            dms::DistributedMessageSet::fetch(node.get_dms(), network_config)
+                .await
+                .unwrap();
+            node.update().await.unwrap();
+        }
     }
-    let _ = serve_task.await.unwrap().unwrap();
-
-    // Todo: verify finalization proofs
-}
-
-#[tokio::test(flavor = "multi_thread")]
-async fn single_seperate_server_propose_1() {
-    setup_test();
-    let (network_id, dms_key) = get_network_id_and_dms_key("single_seperate_server_propose_1");
-
-    let voting_powers = vec![1, 1, 1, 1, 1];
-    let num_nodes = voting_powers.len();
-    let params = ConsensusParams {
-        timeout_ms: 60 * 1_000, // 1 minute
-        repeat_round_for_first_leader: 100,
-    };
-    let round_zero_timestamp = get_timestamp();
 
-    let (server_config, other_configs, peers) =
-        setup_server_client_nodes(network_id.clone(), num_nodes - 1).await;
-
-    let block_header = configs_to_block_header(
-        once(&server_config).chain(&other_configs).collect(),
-        voting_powers,
-    );
-
-    // Create client nodes that will be executed by CLI.
-    let mut nodes = Vec::new();
-    for config in once(&server_config).chain(other_configs.iter()) {
-        let consensus = Consensus::new(
-            create_test_dms(config.clone(), dms_key.clone(), peers.clone()).await,
-            create_storage(create_temp_dir()).await,
-            block_header.clone(),
-            params.clone(),
-            round_zero_timestamp,
-            Some(config.private_key.clone()),
-        )
+    client_nodes[0]
+        .0
+        .set_proposal_candidate(block_hash, 0)
```

Comment:
FYI, L108-112 is actually not needed because the index of the block to propose is initially set to zero in Vetomint. Anyway I think it's better to keep it intact just to be clear and explicit.

Reply:
And I'm not sure if it is intended for Vetomint to have a concrete value "0" as the default block index to propose, even when the Vetomint core hasn't been informed of any valid block hash.

Reply:
You're right. I made an issue #435